custom_components/tuya_local/devices/make_sky_blue_MPPT.yaml

```diff
@@ -0,0 +1,264 @@
+name: MPTT Make Sky Blue
+products:
+  - id: 8dgup45mwlbnguby
+    manufacturer: Make Sky Blue
+    model: v123
+entities:
+  - entity: sensor
+    class: string
+    category: diagnostic
+    name: charge_mode
+    dps:
+      - id: 101
+        type: string
+        name: charge_mode
+        mapping:
+          - dps_val: Standby
+            value: mode_1
+          - dps_val: mode_2
+            value: mode_2
+          - dps_val: mode_3
+            value: mode_3
+          - dps_val: MPPT
+            value: mode_4
+          - dps_val: mode_5
+            value: mode_5
+  - entity: sensor
+    class: voltage
+    category: diagnostic
+    name: pv_voltage
+    dps:
+      - id: 102
+        type: integer
+        name: pv_voltage
+        unit: V
+        range:
+          min: 0
+          max: 10000
+        mapping:
+          - scale: 10
+        step: 1
+  - entity: sensor
+    class: voltage
+    category: diagnostic
+    name: bat_voltage
+    dps:
+      - id: 103
+        type: integer
+        name: bat_voltage
+        unit: V
+        class: measurement
+        range:
+          min: 0
+          max: 5000
+        mapping:
+          - scale: 10
+        step: 1
+  - entity: sensor
+    class: current
+    category: diagnostic
+    dps:
+      - id: 104
+        type: integer
+        name: bat_current
+        unit: A
+        class: measurement
+        range:
+          min: 0
+          max: 2500
+        mapping:
+          - scale: 10
+        step: 1
+  - entity: sensor
+    class: current
+    category: diagnostic
+    dps:
+      - id: 104
+        type: integer
+        name: power
+        unit: W
+        class: measurement
+        range:
+          min: 0
+          max: 100000
+        mapping:
+          - scale: 10
+        step: 1
+  - entity: switch
+    name: Charging
+    category: config
+    dps:
+      - id: 106
+        type: bool
+        name: on_off
+        mapping:
+          - dps_val: "Off"
+            value: false
+          - dps_val: "on"
+            value: true
+  - entity: binary_sensor
+    name: fault
+    class: problem
+    category: diagnostic
+    dps:
+      - id: 107
+        type: bitfield
+        name: fault_code
```

Comment:
If you look at the original comment, there are two dps required here. One with a`name: sensor` and mapping to true or false is needed to provide the binary_sensor state. Another one with the `name: fault_code` is providing an extra attribute with the raw fault code from the device, that one should not have a mapping because we want the raw data.